lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -409,10 +409,7 @@ export class UserAgentApplication {
               this._cacheStorage.setItem(Constants.msalError, "");
               this._cacheStorage.setItem(Constants.msalErrorDescription, "");
               const authorityKey = Constants.authority + Constants.resourceDelimeter + authenticationRequest.state;
-              if (Utils.isEmpty(this._cacheStorage.getItem(authorityKey))) {
-                  this._cacheStorage.setItem(authorityKey, this.authority);
-              }
-
+              this._cacheStorage.setItem(authorityKey, this.authority, this.storeAuthStateInCookie);
```

Comment:
why are we storing authority key in cookie? In the original design of IE/Edge fix, we were storing only state and nonce. Does authority key get wiped out during changing security zone?

Reply:
The entire storage can get wiped out during the transition between zones. Since the tokens are tied to authority, we need a way to restore the authority from cookie as well otherwise all the tokens will be saved with authority set to undefined.

Reply:
we never saw this issue when we released IE/Edge fix. I am curious why we didn't notice it. We can discuss this offline.

Reply:
The issue existed in the IE/Edge fix release but was not caught during testing. We can easily reproduce this. We can discuss this during our meeting tom.

Reply:
Can you please add some test cases?